[Split Build] Support nightly release

PaliC opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #129253
* __->__ #129011

This PR adds the split build to our binaries workflow. Validation for the workflow is done using the PR above in conjunction with https://github.com/pytorch/builder/pull/1876.

Test Workflow: Check CI in the workflow above

PaliC committed:
[Split Build] Support nightly release

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Support nightly release"




[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Support nightly release"




[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Support nightly release"




[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Support nightly release"


Test Workflow: https://github.com/pytorch/pytorch/actions/runs/9601807487/job/26481169277

[ghstack-poisoned]

atalman commented on .circleci/scripts/binary_linux_test.sh:
```diff
@@ -97,8 +97,17 @@ if [[ "$PACKAGE_TYPE" == conda ]]; then
-    retry pip install -q numpy protobuf typing-extensions
+    if [[ "$USE_SPLIT_BUILD" == "true" ]]; then
+      pkg="$(ls -1 /final_pkgs/torch_no_python* | sort |tail -1)"
+      pip install "\$pkg" --index-url "https://download.pytorch.org/whl/\${CHANNEL}/${DESIRED_CUDA}"
```
- 1 pip command should be enough 
- The channel needs to be set to your package s3 path, now its set to: ``https://download.pytorch.org/whl/nightly/cu121``

PaliC committed:
Update on "[Split Build] Support nightly release"


This PR adds the split build to our binaries workflow. Validation for the workflow is done using the PR above in conjunction with https://github.com/pytorch/builder/pull/1876.

Test Workflow: https://github.com/pytorch/pytorch/actions/runs/9601807487 (igonore the cpu jobs. They were added to get better feedback.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Support nightly release"


This PR adds the split build to our binaries workflow. Validation for the workflow is done using the PR above in conjunction with https://github.com/pytorch/builder/pull/1876.

Test Workflow: https://github.com/pytorch/pytorch/actions/runs/9601807487 (igonore the cpu jobs. They were added to get better feedback.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Support nightly release"


This PR adds the split build to our binaries workflow. Validation for the workflow is done using the PR above in conjunction with https://github.com/pytorch/builder/pull/1876.

Test Workflow: https://github.com/pytorch/pytorch/actions/runs/9601807487 (igonore the cpu jobs. They were added to get better feedback.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Support nightly release"


This PR adds the split build to our binaries workflow. Validation for the workflow is done using the PR above in conjunction with https://github.com/pytorch/builder/pull/1876.

Test Workflow: Check CI in the workflow above

[ghstack-poisoned]

atalman reviewed (approved):
lgtm, as discussed change to one line install

PaliC commented:
@pytorchbot merge

PaliC committed:
Update on "[Split Build] Support nightly release"


This PR adds the split build to our binaries workflow. Validation for the workflow is done using the PR above in conjunction with https://github.com/pytorch/builder/pull/1876.

Test Workflow: Check CI in the workflow above

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Support nightly release"


This PR adds the split build to our binaries workflow. Validation for the workflow is done using the PR above in conjunction with https://github.com/pytorch/builder/pull/1876.

Test Workflow: Check CI in the workflow above

[ghstack-poisoned]

PaliC commented:
@pytorchbot merge -f "pull, trunk, and relevant binary jobs are passing"
